core/templates/AppSpec.ts

```diff
@@ -31,38 +31,101 @@ import { UpgradedServices } from 'services/UpgradedServices';
-      new WrittenTranslationsObjectFactory(
-        new WrittenTranslationObjectFactory()));
-  }));
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    var ugs = new UpgradedServices();
-    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
-      $provide.value(key, value);
-    }
-  }));
+describe('App', function() {
+  describe('Generating Constants', function() {
+    beforeEach(angular.mock.module('oppia'));
+    beforeEach(angular.mock.module('oppia', function($provide) {
+      $provide.value(
+        'ParamChangeObjectFactory', new ParamChangeObjectFactory());
+      $provide.value('RuleObjectFactory', new RuleObjectFactory());
+      $provide.value('VoiceoverObjectFactory', new VoiceoverObjectFactory());
+      $provide.value(
+        'WrittenTranslationObjectFactory',
+        new WrittenTranslationObjectFactory());
+      $provide.value(
+        'WrittenTranslationsObjectFactory',
+        new WrittenTranslationsObjectFactory(
+          new WrittenTranslationObjectFactory()));
+    }));
+    beforeEach(angular.mock.module('oppia', function($provide) {
+      var ugs = new UpgradedServices();
+      for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+        $provide.value(key, value);
+      }
+    }));
 
-  var $injector = null;
-  beforeEach(angular.mock.inject(function(_$injector_) {
-    $injector = _$injector_.get('$injector');
-  }));
+    var $injector = null;
+    beforeEach(angular.mock.inject(function(_$injector_) {
+      $injector = _$injector_.get('$injector');
+    }));
 
-  it('should transform all key value pairs to angular constants', function() {
-    for (var constantName in constants) {
-      expect($injector.has(constantName)).toBe(true);
-      expect($injector.get(constantName)).toEqual(constants[constantName]);
-    }
+    it('should transform all key value pairs to angular constants',
+      function() {
+        for (var constantName in constants) {
+          expect($injector.has(constantName)).toBe(true);
+          expect($injector.get(constantName)).toEqual(constants[constantName]);
+        }
+      });
+  });
+  describe('Exception Handler', function() {
+    beforeEach(angular.mock.module('oppia'));
+    beforeEach(angular.mock.module('oppia', function($provide) {
+      $provide.constant('DEV_MODE', false);
+    }));
+    beforeEach(angular.mock.module('oppia', function($provide) {
+      var ugs = new UpgradedServices();
+      for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+        $provide.value(key, value);
+      }
+    }));
+
+    var $exceptionHandler = null;
+    var $log = null;
+    beforeEach(angular.mock.inject(function(
+        _$exceptionHandler_, _$log_) {
+      $exceptionHandler = _$exceptionHandler_;
+      $log = _$log_;
+      spyOn($log, 'error');
+      spyOn(sourceMappedStackTrace, 'mapStackTrace');
+    }));
+    it('should handle non-Error type exceptions gracefully', function() {
```

Comment:
Please add newline before each of your "it" blocks, and above any "conceptual blocks" of logic in the code.  Each block should represent a single "idea". Otherwise it becomes very hard to parse the test when reading.

Reply:
Done!